src/mlagility/helpers/labels.py

```diff
@@ -0,0 +1,80 @@
+import os
+from typing import Dict, List
+
+
+def load_from_file(file_path: str) -> Dict[str, List[str]]:
+    """
+    This function extracts labels from a Python file.
+    Labels must be in the first line of a Python file and start with "# labels: "
+    Each label must have the format "key::value1,value2,..."
+
+    Example:
+        "# labels: author:google test_group:daily,monthly"
+    """
+    # Labels must be in the first line of the python file
+    with open(file_path, encoding="utf-8") as f:
+        first_line = f.readline()
+
+    if "# labels:" in first_line:
+
+        # Convert labels to dict to ensure they are not malformed
+        label_list = first_line.replace("\n", "").split(" ")[2:]
+        label_dict = convert_to_dict(label_list)
+        return label_dict
+    else:
+        return {}
+
+
+def load_from_cache(cache_dir: str, build_name: str) -> Dict[str, List[str]]:
+    """
+    Loads labels from the cache directory
+    """
+
+    labels_dir = os.path.join(cache_dir, "labels")
+    file_path = os.path.join(labels_dir, f"{build_name}.txt")
+    with open(file_path, encoding="utf-8") as f:
+        first_line = f.readline()
+
+    # Convert labels to dict to ensure they are not malformed
+    label_list = first_line.replace("\n", "").split(" ")
+    label_dict = convert_to_dict(label_list)
+    return label_dict
+
+
+def convert_to_dict(label_list: List[str]) -> Dict[str, List[str]]:
```

Comment:
If a function is only ever going to be used inside of this file, suggest prefixing it with `_`

Reply:
prefixed!